diff --git a/codegen/sklearn_wrapper_template.py_template b/codegen/sklearn_wrapper_template.py_template
index 4187a09..e5f857f 100644
--- a/codegen/sklearn_wrapper_template.py_template
+++ b/codegen/sklearn_wrapper_template.py_template
@@ -139,9 +139,6 @@ def _validate_sklearn_args(args: Dict[str, Any], klass: type) -> Dict[str, Any]:
 
 
 class {transform.original_class_name}(BaseTransformer):
-    r"""{transform.estimator_class_docstring}
-    """
-
     def __init__(  # type: ignore
         {transform.estimator_init_signature}
     ) -> None:
